clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy

```diff
@@ -160,9 +162,43 @@ class ServerGroupController {
   @PostFilter("hasPermission(filterObject?.application, 'APPLICATION', 'READ')")
   @PostAuthorize("@authorizationSupport.filterForAccounts(returnObject)")
   @RequestMapping(value = "/serverGroups", method = RequestMethod.GET)
-  List getServerGroupsByApplications(@RequestParam(value = 'applications') List<String> applications,
-                                     @RequestParam(required = false, value = 'cloudProvider') String cloudProvider) {
-    applications.collectMany { summaryList(it, cloudProvider) }
+  List getServerGroups(@RequestParam(required = false, value = 'applications') List<String> applications,
+                       @RequestParam(required = false, value = 'ids') List<String> ids,
+                       @RequestParam(required = false, value = 'cloudProvider') String cloudProvider) {
+    if ((applications && ids) || (!applications && !ids)) {
+      throw new IllegalArgumentException("Provide either 'applications' or 'ids' parameter (but not both)");
+    }
+
+    if (applications) {
+      return getServerGroupsForApplications(applications, cloudProvider)
+    } else {
+      return getServerGroupsForIds(ids)
+    }
+  }
+
+  private List<ServerGroupViewModel> getServerGroupsForApplications(List<String> applications, String cloudProvider) {
+    return applications.collectMany { summaryList(it, cloudProvider) }
+  }
+
+  private List<ServerGroupViewModel> getServerGroupsForIds(List<String> serverGroupIds) {
+    String[][] allIdTokens = serverGroupIds.collect { it.split(':') }
+
+    def invalidIds = allIdTokens.findAll { it.size() != 3 }
+    if (!invalidIds.empty) {
+      throw new IllegalArgumentException("Expected ids in the format <account>:<region>:<name> but got invalid ids: " +
+        invalidIds.collect { it.join(':') }.join(', '))
+    }
+
+    allIdTokens.collect { String[] idTokens ->
+      def (account, region, name) = idTokens
+      try {
+        def serverGroup = getServerGroup(account, region, name)
+        return new ServerGroupViewModel(serverGroup, serverGroup.moniker.cluster, account)
+      } catch (ignored) {
+        log.info("Couldn't get server group ${idTokens.join(':')}")
```

Comment:
Seems like a strange thing to actually ignore?

`log.error("...", e)` (unless there's a good reason to ignore and not `.error` it